Monika After Story/game/zz_backgrounds.rpy

```diff
@@ -0,0 +1,518 @@
+                return self.image_snow_night
+            return self.image_night
+
+        def getRoomForTime(self, weather=None):
+            """
+            Gets the room for the current time
+
+            IN:
+                weather - get the room bg for the time and weather
+                (Default: current weather)
+            """
+            if weather is None:
+                weather = store.mas_current_weather
+            if store.mas_isMorning():
+                return self.getDayRoom(weather)
+            return self.getNightRoom(weather)
+
+        def isChangingRoom(self, old_weather, new_weather):
+            return self.getRoomForTime(old_weather) != self.getRoomForTime(new_weather)
+
+
+
+#Helper methods and such
+init -20 python in mas_background:
+    BACKGROUND_MAP = {}
+    BACKGROUND_RETURN = "Nevermind"
+
+    def loadMBGData():
+        """
+        Loads persistent MASBackground data into the weather map
+
+        ASSUMES: background map is already filled
+        """
+        if store.persistent._mas_background_MBGdata is None:
+            return
+
+        for mbg_id, mbg_data in store.persistent._mas_background_MBGdata.iteritems():
+            mbg_obj = BACKGROUND_MAP.get(mbg_id, None)
+            if mbg_obj is not None:
+                mbg_obj.fromTuple(mbg_data)
+
+    def saveMBGData():
+        """
+        Saves MASBackground data from weather map into persistent
+        """
+        for mbg_id, mbg_obj in BACKGROUND_MAP.iteritems():
+            store.persistent._mas_background_MBGdata[mbg_id] = mbg_obj.toTuple()
+
+
+#START: BG change functions
+init 800 python:
+
+    def mas_setBackground(_background):
+        """
+        Sets the initial bg
+
+        NOTE: We don't handle exit pp's here
+
+        IN:
+            _background:
+                The background we're changing to
+        """
+        global mas_current_background
+        old_background = mas_current_background
+        mas_current_background = _background
+        mas_current_background.entry(old_background)
+
+    def mas_changeBackground(new_background, by_user=None):
+        """
+        changes the background w/o any scene changes
+
+        IN:
+            new_background:
+                The background we're changing to
+
+            by_user:
+                If the user switched the background themselves
+        """
+        if by_user is not None:
+            mas_background.force_background = bool(by_user)
+
+        mas_current_background.exit(new_background)
+        mas_setBackground(new_background)
+
+    #Just set us to the normal room here
+    mas_current_background = None
+    mas_setBackground(mas_background_def)
+
+    #Make sure the bg selector is only available with the right amount of bgs unlocked
+    if len([
+        (mbg_obj.prompt, mbg_obj, False, False)
+        for mbg_id, mbg_obj in mas_background.BACKGROUND_MAP.iteritems()
+        if mbg_obj.unlocked
+        ]
+        ) < 2:
```

Comment:
i guess you're trying for brevity here, but this does a useless list build. I think you should make this simpler i.e:  like
```
unlocked_count = 0
for mbj_obj in mas_background.BACKGROUND_MAP.itervalues():
    unlocked_count += int(mbg_obj.unlocked)

if unlocked_count < 2:
    # etc
```
and since this is in a global store, then it would be actually better to make that for loop in a function in the mas_background store where it returns the unlocked count, then you can do like:
```python
if mas_background.unlockedBG_count() < 2:
    # etc
```